toolkit/tools/scheduler/scheduler.go

```diff
@@ -466,6 +464,52 @@ func updateGraphWithImplicitProvides(res *schedulerutils.BuildResult, pkgGraph *
 	return
 }
 
+// setAssociatedDeltaPaths sets the RpmPath for all of the request's ancillary nodes (both build and run) to the actual
+// RPM paths. A delta node will normally point at the cached RPM path, but we want to point it at the actual RPM if we built it.
+// This function should only be called on delta build nodes.
+func setAssociatedDeltaPaths(res *schedulerutils.BuildResult, pkgGraph *pkggraph.PkgGraph, graphMutex *sync.RWMutex) (err error) {
+	graphMutex.Lock()
+	defer graphMutex.Unlock()
+
+	// Build map of basename to full path for built files. This will allow us to find the actual RPM path we built for
+	// any given .rpm file built from our ancillary nodes.
+	builtFileMap := make(map[string]string)
+	for _, builtFile := range res.BuiltFiles {
+		// We should not expect to see multiple built files with the same basename
+		baseName := filepath.Base(builtFile)
+		_, hasConflict := builtFileMap[baseName]
+		if hasConflict {
+			err = fmt.Errorf("multiple built files with same basename: %s", baseName)
+			return
+		}
+		logger.Log.Tracef("Built delta file: %s", builtFile)
+		builtFileMap[baseName] = builtFile
+	}
+
+	// Now we can scan for all the run nodes that use the cached RPM path and update them to the actual RPM path.
+	for _, node := range pkgGraph.AllRunNodes() {
+		// Get base path of the .rpm for the node and find the built file in the map
+		rpmBasePath := filepath.Base(node.RpmPath)
+		builtFile, ok := builtFileMap[rpmBasePath]
+		if ok {
+			// We only care about nodes that are deltas
+			if node.State == pkggraph.StateDelta {
+				// Update the node to point at the actual RPM path from our map of built files
+				logger.Log.Tracef("Updating delta run node '%s' path from '%s' to '%s'", node, node.RpmPath, builtFile)
+				node.RpmPath = builtFile
+			} else {
+				// Sanity check that any non-delta node has an exact match to the real RPM path
+				if node.RpmPath != builtFile {
```

Comment:
>if node.RpmPath != builtFile { [](http://example.com/codeflow?start=4&length=30)

Now I'd only convert this to an `else if` to save an indent.

Reply:
Done